AcademyChallenge/Data/API/AvatarNetwork.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  AvatarNetwork.swift
+//  AcademyChallenge
+//
+//  Created by Keven Esmael on 03/10/2022.
+//
+
+import Foundation
+import UIKit
+
+
+class AvatarNetwork {
+	
+	func executeNetwork<ResultType: Decodable>(_ call: APIProtocol, _ resultHandler: @escaping (Result<ResultType, Error>) -> Void) {
```

Comment:
This method is already generic enough and doesn't need to be duplicated in the new Network classes. The preferred approach here is to reuse what we already have, making this `executeNetwork` method common through the Network classes. That can be done either by (1) Inheritance (2) Composition.

Reply:
solved